### PR TITLE
feat: sign-up app

### PR DESCRIPTION
**Pull request by Jisay**
# Add new sign-up application

## :rocket: Feature

feat(sign-up): add new application and corresponding module

## :house: Internal

ref: ECPROJ-258

### CODE REVIEW

File: packages/manager/apps/sign-up/src/activity/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import angular from 'angular';
+import '@ovh-ux/sign-up';
+
+import { registerState } from './routing';
+
+const moduleName = 'signUpActivityState';
```
Comment (marie-j) [inline packages/manager/apps/sign-up/src/activity/index.js]: We might want to prefix modules name with `ovh` due to AngularJS

File: packages/manager/modules/sign-up/src/form/activity/activity.controller.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export default class SignUpActivityCtrl {}
```
Comment (marie-j) [inline packages/manager/modules/sign-up/src/form/activity/activity.controller.js]: If controller is empty, the `component` should probably only be declared with a `template`

File: packages/manager/modules/pci/src/projects/project/private-networks/add/add.controller.js [hunk 3]
```diff
@@ -79,6 +79,7 @@ export default class NetworkAddCtrl {
   isVlanAvailable(vlanId) {
     return !find(this.networks, { vlanId });
   }
+
```
Comment (cbourgois) [inline packages/manager/modules/pci/src/projects/project/private-networks/add/add.controller.js]: Seems not related. Revert-it to avoid a new release of pci module.

Comment (Jisay) [reply packages/manager/modules/pci/src/projects/project/private-networks/add/add.controller.js]: I have tried to remove it but it seems that eslint fix it automatically

File: packages/manager/apps/sign-up/src/details/routing.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+export const state = {
+  name: 'sign-up.details',
+  url: 'details',
+  views: {
+    details: {
+      component: 'signUpDetails',
```
Comment (antleblanc) [inline packages/manager/apps/sign-up/src/details/routing.js]: Add `ovh` prefix maybe?

File: packages/manager/apps/sign-up/src/form/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import component from './component';
+
+const moduleName = 'ovhSignUpFormView';
+
+angular
+  .module(moduleName, [
+    'oui',
+  ])
+  .run(/* @ngTranslationsInject:json ./translations */)
+  .run(($translate, ouiStepperConfiguration) => {
```
Comment (antleblanc) [inline packages/manager/apps/sign-up/src/form/index.js]:
```suggestion
  .run(/* @ngInject */ ($translate, ouiStepperConfiguration) => {
```

File: packages/manager/apps/sign-up/src/index.scss [hunk 6]
```diff
@@ -0,0 +1,31 @@
+@import "~bootstrap4/scss/_functions";
```
Comment (antleblanc) [inline packages/manager/apps/sign-up/src/index.scss]: Mix between single and double quote. Prefer single quote :D

File: packages/manager/apps/sign-up/webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      modules: [
+        path.resolve(process.cwd(), './node_modules'),
+        path.resolve(process.cwd(), '../../../../node_modules'),
+      ],
+      mainFields: [
+        'module',
+        'browser',
+        'main',
+      ],
+      symlinks: false,
```
Comment (antleblanc) [inline packages/manager/apps/sign-up/webpack.config.js]: Must be removed.

File: packages/manager/modules/sign-up/package.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "@ovh-ux/sign-up",
```
Comment (antleblanc) [inline packages/manager/modules/sign-up/package.json]: Must be turned into a private package. Not needed to publish it to NPM I guess.

File: packages/manager/modules/sign-up/README.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
+# sign-up
```
Comment (antleblanc) [inline packages/manager/modules/sign-up/README.md]: Could be updated later since the package will be turned into a private one.

File: packages/manager/modules/sign-up/src/form/details/index.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+// templates for ui-select
+import uiSelectTpl from 'ovh-ui-angular/packages/oui-select/src/templates/select.html';
+import uiSelectChoicesTpl from 'ovh-ui-angular/packages/oui-select/src/templates/choices.html';
+import uiSelectMatchTpl from './ui-select-oui-kit/match.tpl.html';
+
+import component from './details.component';
+
+import './details.scss';
+
+const moduleName = 'signUpDetails';
```
Comment (antleblanc) [inline packages/manager/modules/sign-up/src/form/details/index.js]: Missing prefix.

File: packages/manager/modules/sign-up/src/form/identity/component.js [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import template from './identity.html';
+
+export default {
+  name: 'signUpIdentity',
```
Comment (antleblanc) [inline packages/manager/modules/sign-up/src/form/identity/component.js]: Missing prefix.

File: packages/manager/modules/sign-up/src/form/identity/index.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import angular from 'angular';
+import 'ovh-ui-angular';
+import '@ovh-ux/ng-translate-async-loader';
+import 'angular-translate';
+
+import component from './component';
+
+const moduleName = 'signUpIdentity';
```
Comment (antleblanc) [inline packages/manager/modules/sign-up/src/form/identity/index.js]: Missing prefix.

File: packages/manager/modules/sign-up/src/form/index.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import angular from 'angular';
+import '@ovh-ux/ng-translate-async-loader';
+import 'angular-translate';
+
+import activity from './activity';
+import component from './component';
+import details from './details';
+import identity from './identity';
+
+const moduleName = 'signUpForm';
```
Comment (antleblanc) [inline packages/manager/modules/sign-up/src/form/index.js]: Missing prefix.

File: packages/manager/modules/pci/src/projects/project/private-networks/add/add.controller.js (hunk 3 above)
Comment (antleblanc) [reply packages/manager/modules/pci/src/projects/project/private-networks/add/add.controller.js]: Fixed by #978

**Comment by ovh-cds** [comment]
CDS Report sms#3513.0 ■   
* Stage 1
  * Build CA ✔ 
  * Build EU ■   
  * Build US ✔

**Comment by ovh-cds** [comment]
CDS Report telecom-dashboard#3513.0 ■   
* Stage 1
  * Build CA ✔ 
  * Build EU ■   
  * Build US ✔

**Comment by ovh-cds** [comment]
CDS Report sign-up#3513.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✔ 
  * Build US ✔

**Comment by ovh-cds** [comment]
CDS Report vrack#3513.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ■   
  * Build US ■

**Comment by ovh-cds** [comment]
CDS Report pci#3513.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report public-cloud#3513.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report telecom-dashboard#3513.0 ■   
* Stage 1
  * Build CA ✔ 
  * Build EU ■   
  * Build US ✔

**Comment by ovh-cds** [comment]
CDS Report vrack#3513.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ■   
  * Build US ■

**Comment by ovh-cds** [comment]
CDS Report sms#3513.0 ■   
* Stage 1
  * Build CA ✔ 
  * Build EU ■   
  * Build US ✔

**Comment by ovh-cds** [comment]
CDS Report sign-up#3513.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✔ 
  * Build US ✔

**Comment by ovh-cds** [comment]
CDS Report public-cloud#3513.2 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report pci#3513.2 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report sign-up#3561.2 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report sign-up#3563.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report vrack#3563.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✔ 
  * Build US ✔

**Comment by ovh-cds** [comment]
CDS Report installation#3574.1 ■   
* Stage 1
  * Install ■

**Comment by ovh-cds** [comment]
CDS Report sms#3650.1 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report sms#3653.0 ■   
* Stage 1
  * Build CA ✘  
  * Build EU ■   
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report telecom-task#3653.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report pci#3653.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report freefax#3653.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report public-cloud#3653.0 ■   
* Stage 1
  * Build CA ✘  
  * Build EU ■   
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report overthebox#3653.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report sms#3654.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report overthebox#3654.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✘  
  * Build US ■

**Comment by ovh-cds** [comment]
CDS Report freefax#3654.0 ■   
* Stage 1
  * Build CA ✘  
  * Build EU ■   
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report telecom-task#3654.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✘  
  * Build US ✘

**Comment by ovh-cds** [comment]
CDS Report telecom-dashboard#3654.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✘  
  * Build US ■

**Comment by ovh-cds** [comment]
CDS Report sign-up#3654.0 ✘  
* Stage 1
  * Build CA ✘  
  * Build EU ✘  
  * Build US ✘

File: packages/manager/apps/sign-up/src/activity/routing.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
+export const state = {
+  name: 'sign-up.activity',
+  url: 'activity',
```
[the remark this answers is not in the document]
Comment (Jisay) [reply packages/manager/apps/sign-up/src/activity/routing.js]: As ```'sign-up.activity'``` state is a child of ```'sign-up'``` state and the url of that state is ```'/'``` then the url of ```'sign-up.activity'``` will be ```'//activity'```

File: packages/manager/apps/sign-up/src/activity/routing.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+  views: {
+    activity: {
```
[the remark this answers is not in the document]
Comment (Jisay) [reply packages/manager/apps/sign-up/src/activity/routing.js]: Thanks for the tips

File: packages/manager/apps/sign-up/src/assets/theme/_button.less [hunk 16]
```diff
@@ -0,0 +1,133 @@
+@import './_variables.less';
+
+.oui-button {
+  transition: all ease-in 200ms;
```
[the remark this answers is not in the document]
Comment (Jisay) [reply packages/manager/apps/sign-up/src/assets/theme/_button.less]: This is the same in public-cloud manager

[the remark this answers is not in the document]
Comment (antleblanc) [reply packages/manager/apps/sign-up/src/assets/theme/_button.less]: This change should be addressed by the ODS theme I guess.
